ccd_snr/sections/_s2_model.py

```diff
@@ -0,0 +1,425 @@
+absorbed at a depth $z$ into the epitaxial layer.
+The total \CCE\ is then the average differential \CCE\ weighted by 
+the probability of absorbing a photon at a depth $z$,
+\begin{equation} \label{cce}
+    \text{CCE}(\lambda) = \frac{\int_0^\infty \eta(z) \exp(-\alpha z) \, dz}
+                               {\int_0^\infty \exp(-\alpha z) \, dz},
+\end{equation}
+where $\alpha$ is the absorption coefficient of silicon for the given wavelength.
+
+In principle, $\eta(z)$ is a function of the exact implant profile which is
+usually impractical to measure, but see \cite{Stern2004,Boerner2012} for a case 
+where the authors did have a measurement of the exact implant profile provided 
+by the manufacturer.
+In \citet{Stern1994}, the authors instead adopt a piecewise-linear approximation 
+of the differential \CCE,
+\begin{equation} \label{differential-cce}
+    \eta(z) = \begin{cases}
+        \eta_0 + (1 - \eta_0) z / W, & 0 < z < W \\
+        1, & W < z < \infty
+    \end{cases}
+\end{equation}
+where $\eta_0$ is the differential \CCE\ at the back surface of the sensor.
+Plugging Equation \ref{differential-cce} into Equation \ref{cce} yields an
+arithmetic expression for the \CCE,
+\begin{equation}
+    \text{CCE}(\lambda) = \eta_0 + \left( \frac{1 - \eta_0}{\alpha W} \right)(1 - e^{-\alpha W}),
+\end{equation}
+which can be used in Equation \ref{eq:quantum-efficiency} to determine the \QE.
+In Figure \ref{fig:absorbanceAndCCE} we've plotted $\text{CCE}(\lambda)$ for
+the \citet{Heymes2020} parameters in Table \ref{table:models}.
+
+In \citet{Stern1994}, the authors define an effective \QE\ as
+\begin{equation} \label{eqe}
+    \text{EQE}(\lambda) = A(\lambda) \times \text{CCE}(\lambda),
+\end{equation}
+which is the quantity that is typically measured when calibrating a image sensor
+\citep{Stern1994,Stern2004,Boerner2012}.
+In Figure~\ref{fig:eqe}, we've plotted the measured, effective \QE\ for two
+sources: \citet{Boerner2012} which measured the \AIA\ \CCDs\ at a few
+discrete wavelengths, and \citet{Heymes2020} which measured a Teledyne e2v CCD97
+sensor over a wide wavelength range with high resolution using a monochromator.
+
+Using the Nelder-Mead minimization algorithm \citep{Gao2010},
+we found the free parameters of our model which best fit the data in 
+\citet{Boerner2012} and \citet{Heymes2020}.
+These models are plotted as solid lines in Figure~\ref{fig:eqe},
+and the corresponding values of the free parameters are shown in 
+Table~\ref{table:models}.
+Throughout the remainder of this work we will use the model which best fits
+the \citet{Heymes2020} data.
+"""
+    )
+    result.append(subsection_qe)
+
+    subsection_noise = aastex.Subsection("Noise")
+    subsection_noise.append(ccd_snr.figures.noise_photon())
+    subsection_noise.append(ccd_snr.figures.noise_electron())
+    subsection_noise.append(
+        r"""
+Our noise model will consider three sources:
```

Comment:
Since the IQY is so stable, we need a procedure the yields a physically-meaningful number of electrons while still having the correct expectation value. The procedure below is an attempt at the simplest procedure that satisfies these constraints.